app/src/main/java/com/cornellappdev/android/eatery/MenuActivity.java

```diff
@@ -190,8 +214,9 @@ public void onOffsetChanged(AppBarLayout appBarLayout, int verticalOffset) {
     }
 
     // Formatting for when eatery is a dining hall and has a menu
-    else if (cafeData.getIsDiningHall()
-        && !cafeData.getWeeklyMenu().get(0).toString().equals("[]")) {
+    else if (mEatery instanceof DiningHallModel
+        && !((DiningHallModel) mEatery).getMenuForDay(DayOfWeek.of(0)).toString()
+        .equals("[]")) {
```

Comment:
this `if` statement is honestly a tragedy, can you put a TODO that this needs to be refactored

Reply:
I tried to clean it up, added a note.